third_party/move/move-vm/runtime/src/interpreter.rs

```diff
@@ -1450,6 +1506,29 @@ impl FrameTypeCache {
         )?)
     }
 
+    #[inline(always)]
+    fn get_struct_variant_fields_types(
+        &mut self,
+        idx: StructVariantInstantiationIndex,
+        resolver: &Resolver,
+        ty_args: &[Type],
+    ) -> PartialVMResult<&[(Type, NumTypeNodes)]> {
+        Ok(Self::get_or(
+            &mut self.struct_variant_field_type_instantiation,
+            idx,
+            |idx| {
+                Ok(resolver
+                    .instantiate_generic_struct_variant_fields(idx, ty_args)?
```

Comment:
I am curious about performance given that we have a large (1023?) upper bound on the number of variant fields, if I am not mistaken. Should the upper bound be lower?